snippets/geometric_progression.md

```diff
@@ -0,0 +1,24 @@
+---
+title: geometric progression
+tags: math,beginner
+---
+
+Returns a list of numbers in the geometric progression starting with the given positive integer and up to the specified limit.
+
+There general form of geometric sequence is a, ar, ar<sup>2</sup>, ar<sup>3</sup>, ar<sup>4</sup>, ...
+<br> where r -> common ratio and a -> initial value
+
+- Use `while` with the appropriate conditions to run within limits.
+
+```py
+def geometric_progression(a, r, lim):
+  gp = []
+  while a < lim:
```

Comment:
You can most likely use a list comprehension with range instead of the `while` loop.